### PR TITLE
Add deleteConnection method

### PR DESCRIPTION
**Pull request by logangingerich**
https://linear.app/workos/issue/SDK-17/add-delete-connectionsid-for-node-sdk

Adds a `deleteConnection` method to wrap `DELETE /connections/:id` endpoint.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "version": "0.8.4",
+  "version": "0.8.5",
```
Comment (marktran) [inline package.json:2]: Can you break the version bump into a separate PR?

PRs get squashed into a single commit so we want the ability to easily revert functionality.

File: src/sso/sso.spec.ts [hunk 2]
```diff
@@ -258,6 +258,18 @@ describe('SSO', () => {
       });
     });
 
+    describe('deleteConnection', () => {
+      it(`sends request to delete a Connection`, async () => {
```
Comment (marktran) [inline src/sso/sso.spec.ts:262]:
```suggestion
      it('sends request to delete a Connection', async () => {
```

File: package.json (hunk 1 above)
Comment (logangingerich) [reply package.json:2]: @marktran Absolutely, I'll update shortly 👍

Comment (maxdeviant) [reply package.json:2]: Also, since we're adding functionality we should be bumping the minor version to follow [SemVer](https://semver.org/):
> 2. MINOR version when you add functionality in a backwards compatible manner, and

Comment (logangingerich) [reply package.json]: Added a commit reverting package.json version and snapshots back to 0.8.4

Comment (logangingerich) [reply package.json]: @maxdeviant I can plan for a bump to 0.9.0 (sorry, bit of a new area for me 😅 )

**Review by maxdeviant — APPROVED** [review]
